apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -291,26 +304,30 @@ start_link(MgrId, ResId, Group, ResourceType, Config, Opts) ->
         query_mode = maps:get(query_mode, Opts, sync),
         config = Config,
         opts = Opts,
-        status = connecting,
         state = undefined,
         error = undefined
     },
     gen_statem:start_link(?MODULE, {Data, Opts}, []).
 
-init({Data, Opts}) ->
+init({DataIn, Opts}) ->
     process_flag(trap_exit, true),
-    %% init the cache so that lookup/1 will always return something
-    DataWithPid = Data#data{pid = self()},
-    insert_cache(DataWithPid#data.id, DataWithPid#data.group, DataWithPid),
+    Data = DataIn#data{pid = self()},
     case maps:get(start_after_created, Opts, ?START_AFTER_CREATED) of
-        true -> {ok, connecting, DataWithPid, {next_event, internal, start_resource}};
-        false -> {ok, stopped, DataWithPid}
+        true ->
+            %% init the cache so that lookup/1 will always return something
+            UpdatedData = update_state(Data#data{status = connecting}),
+            {ok, connecting, UpdatedData, {next_event, internal, start_resource}};
+        false ->
+            %% init the cache so that lookup/1 will always return something
+            UpdatedData = update_state(Data#data{status = stopped}),
+            {ok, stopped, UpdatedData}
     end.
 
+terminate({shutdown, removed}, _State, _Data) ->
+    ok;
 terminate(_Reason, _State, Data) ->
-    _ = stop_resource(Data),
-    _ = maybe_clear_alarm(Data#data.id),
-    delete_cache(Data#data.id, Data#data.manager_id),
+    _ = maybe_stop_resource(Data),
+    ok = delete_cache(Data#data.id, Data#data.manager_id),
```

Comment:
We don't need `maybe_clear_alarm` anymore?

Reply:
It actually should have already been cleaned in https://github.com/emqx/emqx/blob/c6f67c104389916683035b30dd2d6960f11a63d2/apps/emqx_resource/src/emqx_resource_manager.erl#L547-L548 at this point.